src/components/PopupWithForm.js

```diff
@@ -4,35 +4,45 @@ class PopupWithForm extends Popup {
   constructor(popupSelector, handleFormSubmit) {
     super({ popupSelector });
     this._handleFormSubmit = handleFormSubmit;
-    this._form = document.querySelector(popupSelector).querySelector("form");
+    this._form = this._popupElement.querySelector("form");
     this._inputList = [...this._popupElement.querySelectorAll("input")];
+    this.submitBtn = this._popupElement.querySelector(".modal__save-button");
+    this._submitBtnText = this.submitBtn.textContent;
+
   }
 
   setPreviewedValues(values) {
     this._inputList.forEach((input) => {
       input.value = values[input.name];
+
     });
   }
   _getInputValues() {
     const inputValues = {};
 
-    this._inputList.forEach((input) => {
+    this._inputList.forEach((input) => { // look at this 
       inputValues[input.name] = input.value;
     });
-
     return inputValues;
   }
 
+  open(){
+    super.open();
+    this.submitBtn.textContent = "Save";
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.